frontend/src/components/pages/WaitlistExperience/index.tsx

```diff
@@ -1,15 +1,83 @@
+import { Center, Spinner } from "@chakra-ui/react";
+import React, { useEffect, useRef, useState } from "react";
+import { useParams } from "react-router-dom";
+import CampsAPIClient from "../../../APIClients/CampsAPIClient";
+import { WaitListExperienceCamper } from "../../../types/CamperTypes";
+import { CampResponse, CampSession } from "../../../types/CampsTypes";
+import WaitlistConfirmationPage from "./ConfirmationPage/ConfirmationPage";
 import WaitlistPersonalInfoPage from "./PersonalInfo";
+import { checkPersonalInfoFilled } from "./PersonalInfo/waitListInfoReducer";
 import WaitlistFooter from "./WaitlistFooter";
 
-const WaitlistExperiencePage = (): React.ReactElement => {
-  const [isConfirmationStep, setIsConfirmationStep] = useState(false);
+interface WaitlistExperiencePageProps {
+  campSessions: CampSession[];
+}
+
+const WaitlistExperiencePage = ({
+  campSessions,
+}: WaitlistExperiencePageProps): React.ReactElement => {
+  const { id: campId } = useParams<{ id: string }>();
+  const [camp, setCamp] = useState<CampResponse | undefined>(undefined);
+  const submitBtnRef = useRef<HTMLButtonElement>(null);
+  const [campers, setCampers] = useState<WaitListExperienceCamper[]>([
+    {
+      firstName: "",
+      lastName: "",
+      age: NaN,
+      contact: {
+        firstName: "",
+        lastName: "",
+        email: "",
+        phoneNumber: "",
+        relationshipToCamper: "",
+      }, // Deep copy the contact of the 1st camper's contacts as single source of truth
+    },
+  ]);
+  useEffect(() => {
+    CampsAPIClient.getCampById(campId).then((campResponse) => {
+      if (campResponse.id) {
+        setCamp(campResponse);
+      }
+    });
+  }, [campId]);
 
+  const [isConfirmationStep, setIsConfirmationStep] = useState(false);
   // Temp variables for personal info page
-  const [isPersonalInfoChecked, setIsPersonalInfoChecked] = useState(false);
+  const isPersonalInfoComplete = checkPersonalInfoFilled(campers, camp);
+
+  if (!camp)
+    return (
+      <Center>
+        <Spinner />
+      </Center>
+    );
 
-  const isPersonalInfoComplete = isPersonalInfoChecked;
+  if (!campSessions) {
+    /* eslint-disable-next-line */
+  campSessions=[{
+        id: "123",
+        camp: "12",
+        capacity: 4,
+        campers: [],
+        waitlist: [],
+        dates: [
+          "Thu Jun 30 2022 00:00:00 GMT+0000 (Coordinated Universal Time)",
+          "Fri Jul 1 2022 00:00:00 GMT+0000 (Coordinated Universal Time)",
+        ],
+      },
+      {
+        id: "456",
+        camp: "12",
+        capacity: 3,
+        campers: [],
+        waitlist: [],
+        dates: [
+          "Wed Jun 29 2022 00:00:00 GMT+0000 (Coordinated Universal Time)",
+          "Thu Jul 30 2022 00:00:00 GMT+0000 (Coordinated Universal Time)",
+        ],
+      },
+    ];
+  } // TODO: Remove this in the future when we get camp session data being passed into component
 
   const submitHandler = () => {
```

Comment:
think the submit handler is wrong here for the waitlist case -- we don't call it the personal info step anymore (and the alert is fine by me as a notification, but let's take out the "PLACEHOLDER - " too